### ISSUE TITLE
`mimalloc` returns non-aligned memory

### ISSUE THREAD

**Issue by emilk** [purpleprotocol/mimalloc_rust#128]
I had this code fail (on both Mac and Windows):

```rs
#[global_allocator]
static GLOBAL: AccountingAllocator<mimalloc::MiMalloc> =
    AccountingAllocator::new(mimalloc::MiMalloc);

#[repr(C, align(256))]
pub struct BigStruct { … }

fn collect() {
    assert_eq!(align_of::<BigStruct>(), 256);
    assert_eq!(size_of::<BigStruct>(), 256);
    let vec: Vec<T> = std::iter::once(BigStruct{…}).collect();
    dbg!(vec.as_ptr() as usize % std::mem::align_of::<T>()); // Sometimes prints 64
    let slice = vec.as_slice(); // panics in debug builds because of the unaligned memory
}
```

Granted, 256 is a very large alignment, but I expect either mimalloc to handle it, or to log a warning, or panic, or similar.
But I don't want it to quietly return unaligned memory.
I cannot find any docs saying there is a limit to the supported alignment.

The standard allocator never fails on the same code.

## Maybe related
* https://github.com/microsoft/mimalloc/issues/603

**Comment by nathaniel-daniel** [comment]
I'm assuming this is from https://github.com/rerun-io/rerun

Looks like you've pinned your mimalloc dep to `0.1.37`. The alignment issues were fixed in `0.1.39`. You can also potentially work around that by disabling the `secure` feature (disable default features), but I didn't extensively check to see if that fixed things.

Relevant Issue: https://github.com/purpleprotocol/mimalloc_rust/issues/87

**Comment by emilk** [comment]
Thank you @nathaniel-daniel !

I did run `cargo update -p mimalloc` and when it didn’t update I assumed we were using the latest version, not realizing we had pinned mimalloc 🤦
